lib/flutter_chips_selector.dart

```diff
@@ -321,46 +323,66 @@ class ChipsSelectorState<T> extends State<ChipsSelector<T?>> {
     var size = renderBox?.size;
 
     return OverlayEntry(
-      builder: (context) => Positioned(
-        width: size!.width,
-        child: Visibility(
-          visible: _suggestions.isNotEmpty,
+      builder: (context) {
+        return Positioned(
+          width: size!.width,
           child: CompositedTransformFollower(
             link: this._layerLink,
             showWhenUnlinked: false,
             child: FocusScope(
               child: Material(
                 elevation: 4.0,
-                child: LimitedBox(
-                  //suggestion list shall be one third of the viewport max
-                  maxHeight: MediaQuery.of(context).size.height / 3,
-                  child: ListView.builder(
-                    controller: _overlayScrollController,
-                    shrinkWrap: true,
-                    addAutomaticKeepAlives: true,
-                    padding: const EdgeInsets.symmetric(vertical: 8),
-                    itemCount: _suggestions.length,
-                    itemBuilder: (BuildContext context, int index) {
-                      return Container(
-                        color: _selectedIndex == index
-                            ? Theme.of(context).hoverColor
-                            : Colors.transparent,
-                        child: widget.suggestionBuilder(
-                            context, this, _suggestions[index]),
-                      );
-                    },
+                child: TapRegion(
+                  behavior: HitTestBehavior.opaque,
+                  onTapOutside: (event) {
+                    FocusScope.of(context).unfocus();
+                  },
+                  // Visibility is down here because we still want to use the tapRegion of the overlay
+                  // if we introduce another tap region outside the overlay for unfocusing the textField
+                  // we get the same issues as if we used the default onTapOutside of TextField
+                  child: Visibility(
+                    visible: _overlayIsVisible,
+                    child: LimitedBox(
+                      //suggestion list shall be one third of the viewport max
+                      maxHeight: MediaQuery.of(context).size.height / 3,
+                      child: ListView.builder(
+                        controller: _overlayScrollController,
+                        shrinkWrap: true,
+                        addAutomaticKeepAlives: true,
+                        padding: const EdgeInsets.symmetric(vertical: 8),
+                        itemCount: _suggestionsWithoutItems.length,
+                        itemBuilder: (BuildContext context, int index) {
+                          return Container(
+                            color: _selectedIndex == index ? Theme.of(context).hoverColor : Colors.transparent,
+                            child: widget.suggestionBuilder(context, this, _suggestionsWithoutItems[index]),
+                          );
```

Comment:
It's been like this before but I don't think painting a hover color should be out responsibility here. The supplied `suggestionBuilder` Widget should take care of this IMO what do you think?

Reply:
dito, removed